src/map/clan.c

```diff
@@ -0,0 +1,839 @@
+
+	return 0;
+}
+
+/**
+ * Checks every clan player and remove those who are inactive
+ */
+int clan_inactivity_kick(int tid, int64 tick, int id, intptr_t data)
+{
+	struct clan *c = NULL;
+	int i;
+
+	if ((c = clan->search(id)) != NULL) {
+		if (!c->kick_time || c->tid != tid || tid == INVALID_TIMER || c->tid == INVALID_TIMER) {
+		  ShowError("Timer Mismatch (Time: %d seconds) %d != %d", c->kick_time, c->tid, tid);
+		  return 0;
+		}
+		for (i = 0; i < ARRAYLENGTH(c->member); i++) {
+			if (c->member[i].char_id <= 0 || c->member[i].online <= 0)
+				continue;
+
+			if (c->member[i].online) {
+				struct map_session_data *sd = c->member[i].sd;
+				if (DIFF_TICK32(sockt->last_tick, sd->idletime) > c->kick_time) {
+					clan->leave(sd, false);
+				}
+			} else if ((time(NULL) - c->member[i].last_login) > c->kick_time) {
+				c->member[i].online = 0;
+				c->member[i].sd = NULL;
+				c->member[i].account_id = 0;
+				c->member[i].char_id = 0;
+				c->member[i].last_login = 0;
+
+				clif->clan_onlinecount(c);
+			}
+
+		}
+		c->tid = timer->add(timer->gettick() + c->check_time, clan->inactivity_kick, c->clan_id, 0);
+	}
+	return 1;
+}
+
+/**
+ * Processes any (plugin-defined) additional fields for a clan entry.
+ *
+ * @param[in,out] entry  The destination clan entry, already initialized
+ *                       (clan_id is expected to be already set).
+ * @param[in]     t      The libconfig entry.
+ * @param[in]     n      Ordinal number of the entry, to be displayed in case
+ *                       of validation errors.
+ * @param[in]     source Source of the entry (file name), to be displayed in
+ *                       case of validation errors.
+ */
+void clan_read_db_additional_fields(struct clan *entry, struct config_setting_t *t, int n, const char *source)
+{
+	// do nothing. plugins can do own work
+}
+
+/**
+ * Processes one mobdb entry from the libconfig file, loading and inserting it
+ * into the mob database.
+ *
+ * @param mobt   Libconfig setting entry. It is expected to be valid and it
+ *               won't be freed (it is care of the caller to do so if
+ *               necessary).
+ * @param n      Ordinal number of the entry, to be displayed in case of
+ *               validation errors.
+ * @param source Source of the entry (file name), to be displayed in case of
+ *               validation errors.
+ * @return bool.
+ */
+bool clan_read_db_sub(struct config_setting_t *clans, int n, const char *source)
+{
+	struct config_setting_t *clan_;
+	struct config_setting_t *allies;
+	struct config_setting_t *antagonists;
+	struct config_setting_t *buff;
+	struct clan *c;
+	const char *aName, *aLeader, *aMap, *aConst;
+	int kicktime = 0, kickchecktime = 0, id, i;
+
+	nullpo_retr(false, clans);
+
+	clan_ = libconfig->setting_get_elem(clans, n);
+	allies = libconfig->setting_get_member(clan_, "Allies");
+	antagonists = libconfig->setting_get_member(clan_, "Antagonists");
+
+	if (!libconfig->setting_lookup_int(clan_, "Id", &id)) {
```

Comment:
I see no checks for duplicate IDs and names. Duplicate key entries are directly being added to the db below without any prior checks. For optimal performance and usability you need to cache the data into a temporary local structure and then copy the contents to the heap-allocated memory before adding a pointer to the db.